crates/context/config/src/client/env/config/mutate.rs

```diff
@@ -152,8 +152,11 @@ impl<'a> Method<Icp> for Mutate<'a> {
         } else {
             // Non-empty response means there was an error message
             let error_msg = String::from_utf8(response)
-                .map_err(|e| eyre::eyre!("Invalid UTF-8 in error message: {}", e))?;
-            eyre::bail!("{}", error_msg)
+                .map_err(|e| eyre::eyre!("Invalid UTF-8 in the error response: {}", e))?;
+            eyre::bail!(
+                "Error response received from the system: '{}'",
+                error_msg
+            );
```

Comment:
Errors shouldn't come here at all. The transport should've reported the error directly to the client, at which point it will short-circuit and be returned to the user.

Reply:
So updated this and used same logic as we have in near